tests/common/PlatformInfo.cs

```diff
@@ -187,6 +187,47 @@ public static bool IsAvailable (this IEnumerable<OSPlatformAttribute> attributes
 			// Only available if there aren't any attributes for other platforms
 			// return attributes.Count () == 0;
 		}
+
+		public static bool? IsAvailable (IEnumerable<(OSPlatformAttribute Attribute, ApplePlatform Platform, Version Version)> attributes, PlatformInfo targetPlatform, ApplePlatform attributePlatform)
+		{
+			// First we check for any unsupported attributes, and only once we know that there aren't any unsupported
+			// attributes, we check for supported attributes. Otherwise we might determine that an API is available
+			// if we check the supported attribute first for OS=3.0 here:
+			//     [SupportedOSPlatform ("1.0")
+			//     [UnsupportedOSPlatform ("2.0")
+			// if we run into the SupportedOSPlatform attribute first.
+			foreach (var parsedAttr in attributes) {
+				var attr = parsedAttr.Attribute;
+				var platform = parsedAttr.Platform;
+				var version = parsedAttr.Version;
```

Comment:
You can have nicer code deconstructing the attributes tuple already in the foreach, no need to add the variables:
```suggestion
			foreach (var (attr, platform, version) in attributes) {
				var attr = parsedAttr.Attribute;
				var platform = parsedAttr.Platform;
				var version = parsedAttr.Version;
```